src/Components/DrawingCanvas.jsx

```diff
@@ -0,0 +1,308 @@
+        context.fillStyle = "white";
+      }
+    
+      const strokeSize = () => {
+        const { context } = getContext();
+        context.lineWidth = document.getElementById('lineWidth').value;
+      }
+    
+      const fillCanvas = () => {
+        const { context, drawCanvas } = getContext();
+        const { displayContext } = getContextDisplay();
+        colorPicker();
+        context.fillRect(0, 0, drawCanvas.width, drawCanvas.height);
+        displayContext.drawImage(drawCanvas, 0, 0);
+        drawingHistory.push(drawCanvas.toDataURL());
+        drawingNumber ++;
+      }
+    
+      const resetCanvas = () => {
+        const { context } = getContext();
+        const { displayCanvas, displayContext } = getContextDisplay();
+        drawingHistory.push(displayCanvas.toDataURL());
+        context.reset();
+        displayContext.reset();
+        drawingHistory.push(displayCanvas.toDataURL());
+        drawingNumber ++;
+        colorPicker();
+      }
+    
+      const undo = () => {
+        const { context , drawCanvas } = getContext();
+        const { displayCanvas, displayContext } = getContextDisplay();
+    
+        checkNumber = drawingNumber;
+        console.log(checkNumber, drawingNumber);
+        redoStates.push(displayCanvas.toDataURL());
+    
+        if(drawingHistory.length > 0) {
+          context.clearRect(0, 0, drawCanvas.width, drawCanvas.height);
+          displayContext.clearRect(0, 0, drawCanvas.width, drawCanvas.height);
+          drawingHistory.pop();
+          const img = new Image();
+          img.src = drawingHistory[drawingHistory.length - 1];
+          img.addEventListener("load", () => {
+            displayContext.drawImage(img, 0, 0);
+            context.drawImage(img, 0, 0);
+          });
+        }
+      }
+    
+      const redo = () => {
+        if(drawingNumber > checkNumber) {
+          redoStates = [];
+        }
+    
+        const { context, drawCanvas } = getContext();
+        const { displayContext } = getContextDisplay();
+    
+        drawingHistory.push(drawCanvas.toDataURL());
+    
+        if (redoStates.length > 0) {
+          const img = new Image();
+          img.src = redoStates.pop();
+          img.addEventListener("load", () => {
+            context.clearRect(0, 0, drawCanvas.width, drawCanvas.height);
+            displayContext.clearRect(0, 0, drawCanvas.width, drawCanvas.height);
+            displayContext.drawImage(img, 0, 0);
+            context.drawImage(img, 0, 0);
+          });
+        };
+      };
+
+      window.addEventListener("keydown", (ev) => {
+        if (ev.key === "z" && ev.ctrlKey) {
+            undo();
+        };
+    
+        if (ev.key === "z" && ev.metaKey) {
+            undo();
+        };
+      })
+    
+      window.addEventListener("keydown", (ev) => {
+          if (ev.key === "y" && ev.ctrlKey) {
+              redo();
+          } else if (ev.key === "y" && ev.metaKey) {
+              redo();
```

Comment:
Ambele `if`-uri fac redo. Ai putea să combini condițiile într-una
```suggestion
          if (ev.key === "y" && (ev.ctrlKey || && ev.metaKey)) {
              redo();
```

la fel mai sus la `undo`